src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,189 @@
+            assertWithMessage(checkNameInconsistentErrorMsg)
+                    .that(checkNameInText).matches(checkNameInAttribute);
+            final String checkNameModuleErrorMsg = String.format(Locale.ROOT,
+                    "checkName %s is not in %s module", checkNameInAttribute, moduleName);
+            if (checkNameInAttribute.equals(COMMENTS_INDENTATION)
+                    || checkNameInAttribute.equals(INDENTATION)) {
+                assertWithMessage(checkNameModuleErrorMsg).that(moduleName).matches(MISC);
+            }
+            else if (checkNameInAttribute.equals(SUPPRESS_WARNINGS_HOLDER)) {
+                assertWithMessage(checkNameModuleErrorMsg).that(moduleName).matches(ANNOTATION);
+            }
+            else {
+                final String[] moduleFileNames = checksNamesMap.get(moduleName);
+                final String moduleNameErrorMsg = String.format(Locale.ROOT,
+                        "module name: %s is not exist in %s", moduleName, PACKAGE_NAME);
+                assertWithMessage(moduleNameErrorMsg).that(moduleFileNames).isNotNull();
+                boolean match = false;
+                final String checkNameWithSuffix = checkNameInAttribute + SUFFIX_CHECK_JAVA;
+                for (String filename : moduleFileNames) {
+                    if (checkNameWithSuffix.equals(filename)) {
+                        match = true;
+                        break;
+                    }
+                }
+                assertWithMessage(checkNameModuleErrorMsg).that(match).isTrue();
+            }
+        }
+    }
+
+    public static final class CheckTest extends DefaultHandler {
+
+        public static final String SPLIT_CHECK_NAME_IN_ATTRIBUTE = "#";
+
+        public static final String SPLIT_MODULE_NAME_IN_ATTRIBUTE = "\\.";
+
+        public static final String PREFIX_CONFIG = "config_";
+
+        private static final String NODE_NAME = "a";
+
+        private List<List<String>> checkNamesList;
+
+        private List<String> singleCheckNameList;
+
+        private String currentTag;
+
+        @Override
+        public void startDocument() {
+            checkNamesList = new ArrayList<>();
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName,
+                                 Attributes attributes) {
+            if (qName.equals(NODE_NAME)) {
+                final String[] moduleAndCheckName =
+                        attributes.getValue(0).split(SPLIT_CHECK_NAME_IN_ATTRIBUTE);
+                if (moduleAndCheckName[0].startsWith(PREFIX_CONFIG)) {
+                    singleCheckNameList = new ArrayList<>();
+                    final String moduleNameWithHtml =
+                            moduleAndCheckName[0].substring(PREFIX_CONFIG.length());
+                    final String moduleName =
+                            moduleNameWithHtml.split(SPLIT_MODULE_NAME_IN_ATTRIBUTE)[0];
```

Comment:
it looks like same thing can be done with simple regex (remove start and end of module name
`moduleNameWithHtml.replaceAll((".*config_)|(\\.html.*))`

Reply:
done